### PR TITLE
Python 3.11 support

### PR DESCRIPTION
**Pull request by domna**
I think it would be good to start extending the support matrix to newer versions.

### CODE REVIEW

**Comment by lukaspie** [comment]
Currently, [ifes_apt_tc_data_modeling](https://github.com/atomprobe-tc/ifes_apt_tc_data_modeling) is blocking the tests because [v0.0.9](https://pypi.org/project/ifes-apt-tc-data-modeling/) requires Python <3.11, >=3.8. @mkuehbach could you check if newer python versions could also work for this tool?

**Comment by mkuehbach** [comment]
Hi @lukaspie, in principle that lib should also work for 3.11, at the point when the v0.0.9 was written that was not tested though and thus I switched py311 off. As I was planning to start on Monday anyways on implementing the ATO and APT parser for the FAU Erlangen examples, I would then test this with 3.11 and 3.12 and bump the ifes lib accordingly then

File: pyproject.toml [hunk 1]
```diff
@@ -12,11 +12,12 @@ authors = [
 description = "Extend NeXus for materials science experiment and serve as a NOMAD parser implementation for NeXus."
```
Comment (mkuehbach) [inline pyproject.toml:12]: Extend NeXus for experiments and characterization in Materials Science and Materials Engineering ...

**Review by mkuehbach — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (domna) [reply pyproject.toml:12]:
```suggestion
description = "Extend NeXus for experiments and characterization in Materials Science and Materials Engineering and serve as a NOMAD parser implementation for NeXus."
```

**Pull request merged**